### ISSUE TITLE
Problems loading unlicensed games with redump cue/bin or chd images

### ISSUE THREAD

**Issue by gabberhead** [flyinghead/flycast#906]
I have tested some unlicensed games wirh flycast core on retroarch. a lot a working, but with some of them i have problems. here are the games, which i have problems:
-------------------------------------------------
**Unsupported subtype or pre-postgap:**
Breakers (USA, Japan) (Unl)
Cool Herders (USA) (Unl)
Fast Striker Ver. 1.5 (Europe) (Unl)
Feet of Fury (USA) (Unl)
Gunlord (Europe) (Unl)
Last Hope - Pink Bullets (Europe) (Unl)
Leona's Tricky Adventures (Europe) (Unl)
Neo XYX (Europe) (Unl)
Sturmwind (Europe) (Unl) (Rerelease)
Xeno Crisis (USA, Japan) (En,Ja,Fr,De,Es,It,Nl,Pt) (Unl)

chd only. cue/bin are working. except Sturmwind Unsupported subtype or pre-postgap error
-------------------------------------------------
**Unhandled SH4 exception:**
Sturmwind (Europe) (Unl) (Rerelease)

cue/bin only. perhabs chd also, but chd is getting
-------------------------------------------------
**Music/Sound Distortion:**
Dux Version 1.5 (USA, Japan) (Unl)
Redux - Dark Matters (Europe) (Unl)
Wind and Water - Puzzle Battles (Europe) (En,Ja) (Unl)

chd only. cue/bin are working
-------------------------------------------------
**Fatal: SH4 branch instruction in delay slot:**
Ghost Blade (Europe) (Unl) (Limited Edition)
Pier Solar and the Great Architects (USA) (Unl)

chd and cue/bin
-------------------------------------------------
**Freeze at the beginning at Sega Logo:**
Ganryu (USA, Japan) (Unl)

chd and cue/bin
-------------------------------------------------

chd files was made from the redump images with the newest chdman 0.251 version

**Comment by flyinghead** [comment]
What version of the flycast core are you using?
I tested Breakers and Sturwind CUE and I can load them without issue. Please upgrade to the latest core version and retest.

There's an SH4 exception with Sturmwind so there might be an issue since the CDI works fine.

The Pier Solar issue is known. See #378
The Ghost Blade issue is probably due to copy protection.

**Comment by gabberhead** [comment]
i use the  always the latest.

Unsupported subtype or pre/postgap: i get this only with chd images. cue/bin are working.  but sturmwind its not working. not with chd and not with cue/bin

**Comment by flyinghead** [comment]
chdman supports GDI and CUE GD-ROMs but I'm not sure it can create valid CHDs from MIL-CDs.

See https://github.com/mamedev/mame/issues/8242

**Comment by gabberhead** [comment]
Unsupported subtype or pre/postgap with the chds I get, are the All mil-cds?

**Comment by flyinghead** [comment]
Yes, all unofficial Dreamcast games are MIL-CD (mixed audio+data CD-ROMs) while all official games are GD-ROMs.
You can tell the difference by looking at the cue file.

GD-ROMs have a single density area and a high density area:
```
REM SINGLE-DENSITY AREA
FILE "Alienfront Online (USA) (Track 1).bin" BINARY
  TRACK 01 MODE1/2352
    INDEX 01 00:00:00
...
REM HIGH-DENSITY AREA
FILE "Alienfront Online (USA) (Track 3).bin" BINARY
  TRACK 03 MODE1/2352
    INDEX 01 00:00:00
```
MIL-CDs are multisession CD-ROMs:
```
REM SESSION 01
FILE "Breakers (USA, Japan) (Unl) (Track 01).bin" BINARY
  TRACK 01 AUDIO
    INDEX 01 00:00:00
...
REM SESSION 02
FILE "Breakers (USA, Japan) (Unl) (Track 33).bin" BINARY
  TRACK 33 MODE2/2352
    INDEX 01 00:00:00
```

**Comment by gabberhead** [comment]
But why games like finding teddy or some other unlicensed games are working with chd images. Are they not mil-cds?

**Comment by flyinghead** [comment]
I don't know. Finding Teddy has only one audio track and one data track so that may help.

**Comment by gabberhead** [comment]
here ae some other unlicensed games wich are running fine with chd images with flycast:

Alice Dreams Tournament (US) [UNL]
Alice Sisters (US) [UNL]
Battle Crust (US) [UNL]
Captain Tomaday (US) [UNL]
Elansar (EU-DE) [UNL]
Escapee, The (US-DE) [UNL]
Flashback (US) [UNL]
FX Unit Yuki (US) [UNL]
Hermes (EU) [UNL]
Inhabitants (US) [UNL]
Intrepid Izzy (US) [UNL]
Irides - Master of Blocks (US) [UNL]
Magic Pockets (US) [UNL]
Maqiupai (US-DE) [UNL]
Orion's Puzzle Collection (EU) [UNL]
Philia - The Sequel to Elansar (EU) [UNL]
Satazius Next (US) [UNL]
Wolflame (US) [UNL]
Völgarr the Viking (World) [UNL]

**Comment by flyinghead** [comment]
Ghost Blade and Pier Solar both use a similar copy protection feature and need correct disk media catalog and track ISRC numbers. You can see them in the .CUE file:
```
CATALOG 0731453345927
REM SESSION 01
FILE "Track 01.bin" BINARY
  TRACK 01 AUDIO
    ISRC USPR39640267
    INDEX 01 00:00:00
```

As far as I know there is no way to store this information in CHDv5 files, so even if MIL-CD to CHD conversion is supported, these numbers will be lost and the games won't run. This would require a new CHD format version.

**Comment by gabberhead** [comment]
Ok. Thats understandable for chd files. But cue/bin also not working. Ist this fixable for Cue/bin images?

Do u have investigated the Music distortion on dux, redux and Wind and water? The music distortion also happens only with chd files cue bin the Music is working.
